### PR TITLE
feat: support passive auto discovery

### PR DESCRIPTION
**Pull request by ZhiHanZ**

### CODE REVIEW

File: databend-jdbc/src/main/java/com/databend/jdbc/AbstractDatabendResultSet.java [hunk 1]
```diff
@@ -24,33 +23,12 @@
...
-import java.sql.RowId;
-import java.sql.SQLException;
-import java.sql.SQLFeatureNotSupportedException;
-import java.sql.SQLType;
-import java.sql.SQLWarning;
-import java.sql.SQLXML;
-import java.sql.Statement;
-import java.sql.Time;
-import java.sql.Timestamp;
+import java.sql.*;
```
Comment (flaneur2020) [inline databend-jdbc/src/main/java/com/databend/jdbc/AbstractDatabendResultSet.java:27]: prefer to import the concerete types ?

Comment (ZhiHanZ) [reply databend-jdbc/src/main/java/com/databend/jdbc/AbstractDatabendResultSet.java:27]: I see, the IDE linter optimized the import, which is not make sense.

File: databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java [hunk 2]
```diff
@@ -702,17 +690,42 @@ DatabendClient startQueryWithFailover(String sql, StageAttachment attach) throws
...
+
+    /**
+     * Try to auto discovery the databend nodes it will log exceptions when auto discovery failed and not affect real query execution
+     *
+     * @param client the http client to query on
+     * @param settings the client settings to use
+     */
+    void tryAutoDiscovery(OkHttpClient client, ClientSettings settings) {
+        if (this.autoDiscovery) {
+            if (this.driverUri.enableMock()) {
```
Comment (flaneur2020) [inline databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java:714]: neat mock :+1:

File: databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java [hunk 3]
```diff
@@ -702,17 +690,42 @@ DatabendClient startQueryWithFailover(String sql, StageAttachment attach) throws
...
+            }
+            DatabendNodes nodes = this.driverUri.getNodes();
+            if (nodes != null && nodes.needDiscovery()) {
+                try {
+                    nodes.discoverUris(client, settings);
+                } catch (UnsupportedOperationException e) {
+                    logger.log(Level.WARNING, "Current Query Node do not support auto discovery, close the functionality: " + e.getMessage());
+                    this.autoDiscovery = false;
+                } catch (Exception e) {
+                    logger.log(Level.FINE, "Error auto discovery: " + " cause: " + e.getCause() + " message: " + e.getMessage());
```
Comment (flaneur2020) [inline databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java:725]: first time to find out FINE log level, what's it's normal use case? 

(not an issue at all, just curious 👀)

Comment (ZhiHanZ) [reply databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java:757]: info level will print problem by default, I think it is similar to DEBUG level on golang

**Review by hantmac — APPROVED** [review]
LGTM.

File: databend-jdbc/src/main/java/com/databend/jdbc/DatabendNodes.java [hunk 4]
```diff
@@ -3,23 +3,28 @@
...
 
 public class DatabendNodes implements DatabendNodeRouter {
 
     private AtomicReference<List<URI>> query_nodes_uris;
     protected final AtomicInteger index;
     // keep track of latest discovery scheduled time
     protected final AtomicReference<Long> lastDiscoveryTime = new AtomicReference<>(0L);
+    private static final Logger logger = Logger.getLogger(DatabendNodes.class.getPackage().getName());
+    @Setter
+    private boolean debug = false;
```
Comment (flaneur2020) [inline databend-jdbc/src/main/java/com/databend/jdbc/DatabendNodes.java:27]: i guess if there is only one setter, define a `setDebug` is preferrable than using lomok

**Review by flaneur2020 — APPROVED** [review]
LGTM :+1:, only one small style issue about lomok, it's not important

**Pull request merged**